core-js/src/main/javascript/cdf-legacy/components/filter/js/TreeFilter/views/scrollbar/MCustomScrollBarEngineImplementation.js

```diff
@@ -0,0 +1,40 @@
+/*!
+ * Copyright 2002 - 2016 Webdetails, a Pentaho company. All rights reserved.
+ *
+ * This software was developed by Webdetails and is provided under the terms
+ * of the Mozilla Public License, Version 2.0, or any later version. You may not use
+ * this file except in compliance with the license. If you need a copy of the license,
+ * please go to http://mozilla.org/MPL/2.0/. The Initial Developer is Webdetails.
+ *
+ * Software distributed under the Mozilla Public License is distributed on an "AS IS"
+ * basis, WITHOUT WARRANTY OF ANY KIND, either express or implied. Please refer to
+ * the license for the specific language governing your rights and limitations.
+ */
+/**
+ * @summary MCustomScrollBarEngineImplementation implementation of ScrollBarHandler
+ * @description MCustomScrollBarEngineImplementation implementation of ScrollBarHandler
+ */
+(function(AbstractScrollBarHandler, $){
+  var view;
+  var scrollbar;
+  MCustomScrollBarEngineImplementation = AbstractScrollBarHandler.extend({
+    constructor: function (reference) {
+      view = reference;
```

Comment:
With this implementation, if you have two filters in a dashboard, this will not work, because the variables `view`
and `scrollbar` will be overwritten on every instantiation of the `MCustomScrollBarEngineImplementation` class.

I suppose you wanted to have `view` and `scrollbar` as properties of the `MCustomScrollBarEngineImplementation` class. If so, you  should have written:

```javascript
(function(AbstractScrollBarHandler, $){
  MCustomScrollBarEngineImplementation = AbstractScrollBarHandler.extend({
   view: null, // no real need to declare this beforehand
   scrollbar: null, // no real need to declare this beforehand
    constructor: function (reference) {
      this.view = reference; // it might not be necessary to keep a reference to the view 
      var view = reference;
      var options = $.extend(true, {}, view.config.view.scrollbar.options, {
        callbacks: {
          onTotalScroll: function () {
            return view.trigger('scroll:reached:bottom', view.model);
          },
          onTotalScrollBack: function () {
            return view.trigger('scroll:reached:top', view.model);
          }
        }
      });
      this.scrollbar = view.$(view.config.view.slots.children).parent().mCustomScrollbar(options);
      // a constructor should not return anything
    },
    scrollToPosition: function(position) {
      this.scrollbar.mCustomScrollbar("scrollTo",position);
    }
  });
})(AbstractScrollBarHandler, $);

```